docs/DEVELOPING.md

```diff
@@ -737,6 +737,188 @@ $ make ftest NAME=mssql
+
+- The server host url where the SharePoint is hosted
+- Username for SharePoint Server
+- Password for SharePoint Server
+- Client Id for SharePoint Online
+- Client Secret for SharePoint Online
+- Tenant Name for SharePoint Online
+- Tenant Id for SharePoint Online
+- SSL certificate for a secure connection
+
+#### Configure SharePoint connector
+
+The following configuration fields need to be provided for setting up the connector:
+
+Note: The given defaults are  for [E2E](https://github.com/elastic/connectors-python/tree/main/connectors/sources/tests/fixtures/sharepoint) testing purposes only
+
+
+##### `is_cloud`
+
+Flag to determine the SharePoint platform type. `True` if SharePoint cloud and `False` if SharePoint Server. Default value is `False`.
+
+##### `username`
+
+The username of the account for SharePoint Server. Default value is `demo_user`.
+
+Note: `username` is not needed for SharePoint Online.
+
+##### `password`
+
+The password of the account to be used for the SharePoint Server. Default value is `abc@123`.
+
+Note: `password` is not needed for SharePoint Online.
+
+##### `client_id`
+
+The client id to authenticate with SharePoint Online.
+
+##### `client_secret`
+
+The secret id to authenticate with SharePoint Online.
+
+##### `tenant`
+
+The tenant name to authenticate with SharePoint Online.
+
+##### `tenant_id`
+
+The tenant id to authenticate with SharePoint Online.
+
+##### `host_url`
+
+The server host url where the SharePoint is hosted. Default value is `http://127.0.0.1:8491`. Examples:
+
+  - `https://192.158.1.38:8080`
+  - `https://test_user.sharepoint.com`
+
+##### `site_collections`
+
+The site collections to fetch sites from SharePoint(allow comma seprated collections also). Default value is `collection1`. Examples:
+  - `collection1`
+  - `collection1, collection2`
+
+##### `ssl_enabled`
+
+Whether SSL verification will be enabled. Default value is `False`.
+
+##### `ssl_ca`
+
+Content of SSL certificate needed for SharePoint Online. 
+
+Note: Keep this field empty, if `ssl_enabled` is set to `False`. Example certificate:
+
+
+  - ```
+    -----BEGIN CERTIFICATE-----
+    MIID+jCCAuKgAwIBAgIGAJJMzlxLMA0GCSqGSIb3DQEBCwUAMHoxCzAJBgNVBAYT
+    ...
+    AlVTMQwwCgYDVQQKEwNJQk0xFjAUBgNVBAsTDURlZmF1bHROb2RlMDExFjAUBgNV
+    -----END CERTIFICATE-----
+    ```
+
+##### `retry_count`
+
+The number of retry attempts after failed request to the SharePoint. Default value is `3`.
+
+##### `enable_content_extraction`
+
+Whether the connector should extract the content from SharePoint attachments. Default value is `True` i.e. the connector will try to extract file contents.
+
+#### Content Extraction
+
+The connector uses the Elastic ingest attachment processor plugin for extracting file contents. The ingest attachment processor extracts files by using the Apache text extraction library Tika. Supported file types eligible for extraction can be found as `TIKA_SUPPORTED_FILETYPES` in [utils.py](../connectors/utils.py) file.
```

Comment:
We could link to https://www.elastic.co/guide/en/enterprise-search/current/ingest-pipelines.html from this section.